### ISSUE TITLE
Use hashed value of promoteId, when it's a negative stringified BigInt

### ISSUE THREAD

**Issue by stl-vitalinedzvedz** [mapbox/mapbox-gl-js#9788]
## Motivation
In some cases, we use stringified BigInts as feature ids with `promoteId` to use hover styling with feature state. It works well when used with positive stringified BigInts, but breaks with negative ones. In our case, that leads to collisions and multiple features are highlighted on hover.

## Design
A new `promoteId` feature was added to allow the use of feature properties as ids to handle feature state updates like hover ([PR](https://github.com/mapbox/mapbox-gl-js/pull/8987/files#)). There was added a new function [getNumericId](https://github.com/mapbox/mapbox-gl-js/blob/master/src/data/feature_position_map.js#L87-L95), which does coercion to a number if it's not a NaN or less then MAX_SAFE_INTEGER. That check misses case when BigInt is a negative one and function returns an incorrect value instead of a hash.

### Implementation
`getNumericId` could be updated to compare with absolute value of id or could use [Number.isSafeInteger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger):
```
function getNumericId(value: mixed) {
    const numValue = +value;
    if (!isNaN(numValue) && Math.abs(numValue) <= MAX_SAFE_INTEGER) {
        return numValue;
    }
    return murmur3(String(value));
}
```

**Comment by ryanhamley** [comment]
`Number.isSafeInteger` doesn't work in IE but that should be ok soon. @mourner what do you think?

**Comment by mourner** [comment]
@stl-vitalinedzvedz good catch — I think just adding `numValue >= MIN_SAFE_INTEGER` check is good enough as a fix.
